homeassistant/components/boiler/__init__.py

```diff
@@ -0,0 +1,496 @@
+"""
+Provides functionality to interact with boiler/geyser controller units.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/boiler/
+"""
+import logging
+import os
+from numbers import Number
+import voluptuous as vol
+
+from homeassistant.helpers.entity_component import EntityComponent
+
+from homeassistant.config import load_yaml_config_file
+from homeassistant.util.temperature import convert as convert_temperature
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA  # noqa
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_TEMPERATURE, STATE_ON, STATE_OFF, STATE_UNKNOWN,
+    # TEMP_CELSIUS
+    )
+
+DOMAIN = "boiler"
+
+BOILER_CLASSES = [
+    None,                   # Default - electric?
+    'electric',             # Normal electric element
+    'heatpump',             # Uses heatpump as element
+    'on demand electric',   # Probably don't have/need controllers
+    'on demand gas',        # Probably don't have/need controllers
+    'pv solar',             # Uses DC Element
+    'pumped solar',         # Pumps between tank and solar collector
+    'thermosiphon solar',   # Siphons between tank and solar collector
+]
+
+BOILER_CLASSES_SCHEMA = vol.All(vol.Lower, vol.In(BOILER_CLASSES))
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+SCAN_INTERVAL = 60  # Every hour
+
+SERVICE_SET_AWAY_MODE = "set_away_mode"  # trigger to lower target water temp
+SERVICE_SET_GUEST_MODE = "set_guest_mode"  # trigger to raise min temp
```

Comment:
never mind, just saw that guests have total number of guests and holiday has a duration.